### ISSUE TITLE
1.7 update causes the plugin task to crash

### ISSUE THREAD

**Issue by ChaitanyaPramod** [uber/phabricator-jenkins-plugin#71]
I was really looking forward to the 1.7 update. However, the job started failing due to the following. It was fine after I reverted back to 1.6.1 version of the plugin.

1.
```
FATAL: Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory
java.lang.NoClassDefFoundError: Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory
	at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:912)
	at com.uber.jenkins.phabricator.conduit.ConduitAPIClient.perform(ConduitAPIClient.java:67)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.callConduit(DifferentialClient.java:113)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.fetchDiff(DifferentialClient.java:62)
	at com.uber.jenkins.phabricator.PhabricatorBuildWrapper.setUp(PhabricatorBuildWrapper.java:91)
	at hudson.model.Build$BuildExecution.doRun(Build.java:156)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Publisher 'Post to Phabricator' aborted due to exception: 
java.lang.NoClassDefFoundError: Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory
	at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:912)
	at com.uber.jenkins.phabricator.conduit.ConduitAPIClient.perform(ConduitAPIClient.java:67)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.callConduit(DifferentialClient.java:113)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.fetchDiff(DifferentialClient.java:62)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.perform(PhabricatorNotifier.java:122)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:726)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:671)
	at hudson.model.Run.execute(Run.java:1766)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
Finished: FAILURE
```
2.
```
FATAL: INSTANCE
java.lang.NoSuchFieldError: INSTANCE
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.<clinit>(SSLConnectionSocketFactory.java:144)
	at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:912)
	at com.uber.jenkins.phabricator.conduit.ConduitAPIClient.perform(ConduitAPIClient.java:67)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.callConduit(DifferentialClient.java:113)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.fetchDiff(DifferentialClient.java:62)
	at com.uber.jenkins.phabricator.PhabricatorBuildWrapper.setUp(PhabricatorBuildWrapper.java:91)
	at hudson.model.Build$BuildExecution.doRun(Build.java:156)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Publisher 'Post to Phabricator' aborted due to exception: 
java.lang.NoClassDefFoundError: Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory
	at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:912)
	at com.uber.jenkins.phabricator.conduit.ConduitAPIClient.perform(ConduitAPIClient.java:67)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.callConduit(DifferentialClient.java:113)
	at com.uber.jenkins.phabricator.conduit.DifferentialClient.fetchDiff(DifferentialClient.java:62)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.perform(PhabricatorNotifier.java:122)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:726)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:671)
	at hudson.model.Run.execute(Run.java:1766)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
Finished: FAILURE
```

I can't explain this though, as the Apache HTTP client is already a [dependency](https://github.com/uber/phabricator-jenkins-plugin/blob/master/pom.xml#L48-L52).

**Comment by sectioneight** [comment]
D'oh, that is annoying. Can you share your Jenkins version and JRE
version/distro to help reproduce? I'll look at this ASAP

**Comment by sectioneight** [comment]
This appears to be a bug in Jenkins itself:

https://issues.jenkins-ci.org/browse/JENKINS-27289

I have successfully reproduced on 1.609.1. It appears to be fixed in 1.609.2

**Comment by sectioneight** [comment]
Thanks for the report!

Unfortunately the class loader behaves differently in development and testing from when you actually run it on a jenkins slave. I have verified in a VM that the attached PR fixes the issue. I'll cut a 1.7.1 release as soon as it's merged (~10 minutes for TravisCI).

**Comment by ChaitanyaPramod** [comment]
> It appears to be fixed in 1.609.2

Interestingly I'm using Jenkins 1.624 (with Oracle Java 1.8.0_51).

That said, I'll try 1.7.1 version when it's up and report back.

**Comment by ChaitanyaPramod** [comment]
Reporting fixed in 1.7.2. Thanks :+1:

**Comment by sectioneight** [comment]
Sweet. Sorry for the interruption.
